astropy/modeling/astro_sherpa.py

```diff
@@ -0,0 +1,433 @@
+    '''
+    Parameters
+        ----------
+        n_dim: int
+            Used to veirfy required number of dimentions.
+        x : array
+            input coordinates
+        y : array
+            input coordinates
+        z : array (optional)
+            input coordinates
+        xerr : array (optional)
+            an array of errors in x
+        yerr : array (optional)
+            an array of errors in y
+        zerr : array (optional)
+            an array of errors in z
+        n  : int
+            used in error reporting
+
+    returns:
+        _data: a sherpa dataset
+    '''
+
+    if (z is None and n_dim > 1) or (z is not None and n_dim == 1):
+        raise Exception("Model and data dimentions don't match in dataset %i" % n)
+
+    if z is None:
+        assert x.shape == y.shape, "shape of x and y don't match in dataset %i" % n
+    else:
+        z = np.asarray(z)
+        assert x.shape == y.shape == z.shape, "shapes x,y and z don't match in dataset %i" % n
+
+    if yerr is not None:
+        yerr = np.array(yerr)
+        assert y.shape == yerr.shape, "y's and yerr's shapes do not match in dataset %i" % n
+
+    if xerr is not None:
+        xerr = np.array(xerr)
+        assert x.shape == xerr.shape, "x's and xerr's shapes do not match in dataset %i" % n
+
+    if z is not None and zerr is not None:
+        zerr = np.array(zerr)
+        assert z.shape == zerr.shape, "z's and zerr's shapes do not match in dataset %i" % n
+
+    '''
+    So the generall method is to dstack(and transpose each row) so
+        [x1,x2,x3],[y1,y2,y3],[z1,z2,z3],...
+    becomes
+        [x1,y1,,z1],[x2,y2,,z2],[x3,y3,,z3],...
+        the and put it into a data set! this assumes that all columns are set for all datasets!
+        athough a none can be put in place of missing columns
+    '''
+    if z is None:
+        if xerr is None:
+            if yerr is None:
+                _data = Data1D("wrapped_data", x=x, y=y)
+            else:
+                _data = Data1D("wrapped_data", x=x, y=y, staterror=yerr)
+        else:
+            if yerr is None:
+                _data = Data1DInt("wrapped_data", xlo=x - xerr, xhi=x + xerr, y=y)
+            else:
+                _data = Data1DInt("wrapped_data", xlo=x - xerr, xhi=x + xerr, y=y, staterror=yerr)
+    else:
+        if xerr is None and yerr is None:
+            if zerr is None:
+                _data = Data2D("wrapped_data", x0=x, x1=y, y=z)
+            else:
+                _data = Data2D("wrapped_data", x0=x, x1=y, y=z, staterror=zerr)
+        elif xerr is not None and yerr is not None:
+            if zerr is None:
+                _data = Data2DInt("wrapped_data", x0lo=x - xerr, x0hi=x + xerr, x1lo=y - yerr, x1hi=y + yerr, y=z)
+            else:
+                _data = Data2DInt("wrapped_data", x0lo=x - xerr, x0hi=x + xerr, x1lo=y - yerr, x1hi=y + yerr, y=z, staterror=zerr)
+        else:
+            raise Exception("Set xerr and yerr, or set neither!")
```

Comment:
This line is not covered by the tests. Also, you might want to raise a more specific exception.